File: Algebra/Prime Numbers/SieveOfEratosthenes/SieveOfEratosthenes.py
class SieveOfEratosthenes:
    def __init__(self):
        self.primes = []
        self.prime = []

    def sieveOfEratosthenes(self, n):
        self.prime = [True for i in range(n + 1)]

        p = 2
        while p * p <= n:
            if self.prime[p] == True:
                for i in range(p * p, n + 1, p):
                    self.prime[i] = False

            p += 1

        for p in range(2, n + 1):
            if self.prime[p]:
                self.primes.append(p)
        print(self.prime)
        return self.primes


var = SieveOfEratosthenes()
ans = var.sieveOfEratosthenes(50)
print(ans)

File: Algebra/Prime Numbers/SieveOfEratosthenes/SieveOfEratosthenes.java
package SieveOfEratosthenes;
import java.util.ArrayList;

public class SieveOfEratosthenes {

    public ArrayList<Integer> sieveOfEratosthenes(int n) {
        ArrayList<Integer> primes = new ArrayList<>();
        boolean[] prime = new boolean[n+1];
        for(int i=0; i<=n; i++){
            prime[i] = true;
        }
        for (int p = 2; p*p<=n; p++){
            if (prime[p]) {
                for(int i=p*p; i<=n; i+=p){
                    prime[i] = false;
                }
            }
        }
        for (int i = 2; i <= n; i++) {
            if (prime[i]){
                primes.add(i);
            }
        }
        return primes;
    }

    public static void main(String[] args) {
        SieveOfEratosthenes var = new SieveOfEratosthenes();
        int n = 50;
        ArrayList<Integer> ans;
        ans = var.sieveOfEratosthenes(n);
        System.out.println(ans);
    }
}